src/config.rs

```diff
@@ -989,32 +1014,83 @@ impl Config {
             .context("Failed to load config file")?
             .unwrap_or_default();
 
-        Ok(Self::from_env_and_file_configs(env_conf, file_conf))
+        Self::from_env_and_file_configs(env_conf, file_conf)
     }
 
-    fn from_env_and_file_configs(env_conf: EnvConfig, file_conf: FileConfig) -> Self {
+    fn from_env_and_file_configs(env_conf: EnvConfig, file_conf: FileConfig) -> Result<Self> {
         let mut conf_caches: CacheConfigs = Default::default();
 
         let FileConfig {
             cache,
             dist,
             server_startup_timeout_ms,
+            basedirs: file_basedirs,
         } = file_conf;
         conf_caches.merge(cache);
 
         let server_startup_timeout =
             server_startup_timeout_ms.map(std::time::Duration::from_millis);
 
-        let EnvConfig { cache } = env_conf;
+        let EnvConfig {
+            cache,
+            basedirs: env_basedirs,
+        } = env_conf;
         conf_caches.merge(cache);
 
+        // Environment variable takes precedence over file config
+        let basedirs_raw = if !env_basedirs.is_empty() {
+            env_basedirs
+        } else {
+            file_basedirs
+        };
+
+        // Validate that all basedirs are absolute paths
+        // basedirs_raw is Vec<PathBuf>
+        let mut basedirs = Vec::with_capacity(basedirs_raw.len());
+        for p in basedirs_raw {
+            if !p.is_absolute() {
+                bail!("Basedir path must be absolute: {:?}", p);
+            }
+            // Normalize basedir:
+            // remove double separators, cur_dirs
+            let abs = absolute(&p)?;
+            let mut bytes = abs.to_string_lossy().into_owned().into_bytes();
+
+            // remove trailing slashes
+            while matches!(bytes.last(), Some(b'/' | b'\\')) {
+                bytes.pop();
+            }
+            // normalize windows paths
+            let normalized = {
+                #[cfg(target_os = "windows")]
+                {
+                    normalize_win_path(&bytes)
+                }
+
+                #[cfg(not(target_os = "windows"))]
+                {
+                    bytes
+                }
+            };
+            basedirs.push(normalized);
```

Comment:
The tests for trailing slashes must be implemented here for checking how it works with `strip_basedirs`